backend/data_preparation/dumper/fire_dumper.py

```diff
@@ -0,0 +1,438 @@
+        :param aggregated_record: tuple
+                e.g. ("FireA",True,"USFA","California",999,minTime,maxTime,geom....,total_area)
+        :param fire_id: int
+                e.g. 998
+        :return: Tuple[Dict[str, Any]]
+                e.g. ({"name":"FireA"...}, {"name": "FireB"...})
+        """
+        # info = {"name": aggregated_record[0],
+        #         #         "if_sequence": aggregated_record[1],
+        #         #         "agency": aggregated_record[2],
+        #         #         "state": aggregated_record[3],
+        #         #         "id": id,
+        #         #         "start_time": aggregated_record[5],
+        #         #         "end_time": aggregated_record[6],
+        #         #         "geom_full": aggregated_record[7],
+        #         #         "geom_1e4": aggregated_record[8],
+        #         #         "geom_1e3": aggregated_record[9],
+        #         #         "geom_1e2": aggregated_record[10],
+        #         #         "geom_center": aggregated_record[11],
+        #         #         "max_area": aggregated_record[12]
+        #         #         }
+        columns = ["name", "if_sequence", "agency", "state", "id", "start_time", "end_time","geom_full", "geom_1e4",
+                   "geom_1e3", "geom_1e2", "geom_center", "max_area"]
+        info = dict(zip(columns, aggregated_record))
+        info["id"] = fire_id
+        fire_record_update = [info["id"], info["name"], info["start_time"], info["end_time"]]
+        fire_merged_insert = [i for i in info.values()]
+        return fire_record_update, fire_merged_insert
+
+    def get_aggregated_fire_with_id(self, year: int, name: str, state: str, id: int) -> List[Tuple]:
+        """
+        Merges fire events with this id from fire table into several big fire events.
+        Some pages might have fires from multiple fire events, so the return value is a list of tuples.
+        If there are more than one fire events, then there will be multiple tuples in returned list.
+        :param year: int
+                e.g. 1999
+        :param name: str
+                e.g. "FireA"
+        :param state: str
+                e.g. "California"
+        :param id: int
+                e.g. 9999
+        :return: list of tuples representing fire events
+                e.g. [(1999, "FireA",...), (1999, "FireB",....)]
+        """
+        aggregated_fire_records_with_id = list(Connection.sql_execute(self.SQL_GET_LATEST_AGGREGATION.format(id)))
+        if not aggregated_fire_records_with_id:
+            logger.warning(f"Record {id} is an empty record. Skipping...")
+            # if this id is an empty record, then there is no aggregated fire records
+            # in this situation, we only mark the url as crawled, by inserting it into fire_history
+            self.insert_history(FireEvent(year, state, name, id))
+            # return the latest fire id
+            raise InvalidRecordError
```

Comment:
try to pass an error message when raising an error. Then out-scope when catch and handle such error, log it to `logger.error`